dependencies.yaml

```
@@ -88,7 +96,7 @@ dependencies:
           - notebook>=0.5.0
       - output_types: [conda]
         packages:
-          - cugraph=23.08
+          - cugraph==23.8.*
```

Should this be?

Also see other places we might want to make this change

```suggestion
          - cugraph==23.08.*
```

---

No, we want PEP440 compatible pinnings here. https://github.com/rapidsai/cuxfilter/pull/499#discussion_r1258318646

We want to use normalized versions/pinnings that are compatible with both conda and pip. The current state `==23.8.*` is how we do that. See also [cudf's dependencies](https://github.com/rapidsai/cudf/blob/3c51c9ecb56c7130f300bcb21d4ca38a62cbd418/dependencies.yaml#L216-L217) and the format used by `update-version.sh` in various RAPIDS repos.

---

The places where we use "23.08" are places like `__version__`, `CMakeLists.txt`, `README.md`, Doxygen, etc. We have adopted the normalized convention for all conda/pip pinnings.

Compare the usage of `NEXT_SHORT_TAG_PEP440` to the use of `NEXT_SHORT_TAG` in cudf, for example: https://github.com/rapidsai/cudf/blob/branch-23.08/ci/release/update-version.sh

---

Ok sounds good. Wasn't obvious whether the change was intended

---

Hey @jakirkham, it was an intended change based on one of the suggestions by @bdice :)